runtime/lua/vim/lsp/_completion.lua

```diff
@@ -124,39 +125,77 @@ local function adjust_start_col(lnum, line, items, encoding)
   end
 end
 
+---@param line string line content
+---@param lnum integer 0-indexed line number
+---@param client_start_boundary integer 0-indexed word boundary
+---@param server_start_boundary? integer 0-indexed word boundary, based on textEdit.range.start.character
+---@param result lsp.CompletionList|lsp.CompletionItem[]
+---@param encoding string
+---@return table[] matches
+---@return integer? server_start_boundary
+function M._convert_results(
```

Comment:
Nit: Add a `@private` annotation for LuaLS to emit a warning for external callers.